assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -846,8 +846,9 @@ public Assignment addDuplicateAssignment(String context, String assignmentId) th
                 log.debug("Created duplicate assignment {} from {}", assignment.getId(), assignmentId);
 
                 String reference = AssignmentReferenceReckoner.reckoner().assignment(assignment).reckon().getReference();
+                
                 // event for tracking
-                eventTrackingService.post(eventTrackingService.newEvent(AssignmentConstants.EVENT_ADD_ASSIGNMENT, reference, true));
+                eventTrackingService.newEvent(AssignmentConstants.EVENT_ADD_ASSIGNMENT, reference, true);
```

Comment:
In duplicateAssignment the assignment is saved as draft. No need to do the post.

Reply:
You have to change BullhornServiceImpl to avoid generate messages in those cases and don't remove event launch that is right.
https://github.com/sakaiproject/sakai/blob/master/portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java#L287